### PR TITLE
Fixes #785 view and presenter for individual

### LINKED ISSUE #785: View and presenter for IndividualBB
Very similar to StartValue-type presenters and views

This part is not needed from StartValue view
199517077-ed19ec77-25f4-40dc-a0eb-c3214857bb2f

199517139-486bbef1-e991-4265-ab75-584edfbf4fce

Some metadata will be needed as read only

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: src/MoBi.UI/Views/IndividualBuildingBlockView.cs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+using OSPSuite.Presentation.Extensions;
+using OSPSuite.UI.Controls;
+using OSPSuite.UI.Extensions;
+using OSPSuite.UI.RepositoryItems;
+using OSPSuite.Utility.Extensions;
+using static MoBi.Assets.AppConstants.Captions;
+
+namespace MoBi.UI.Views
+{
+   public partial class IndividualBuildingBlockView : BaseUserControl, IIndividualBuildingBlockView
```
Comment (rwmcintosh) [inline src/MoBi.UI/Views/IndividualBuildingBlockView.cs:31]: In the end, refactoring this view to get common code pulled up while maintaining a non-generic base class was unsuccessful. I just couldn't justify any more time on it.

Maybe you have a tip that's going to unlock it for me, but I just couldn't find it.

**Comment by rwmcintosh** [comment]
View uses flow layout to control the variable number of origin data that need to be displayed. Wraps nicely for monitor size and for scaling

image
image

File: src/MoBi.Presentation/DTO/OriginDataItemDTO.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using OSPSuite.Core.Domain.Builder;
+
+namespace MoBi.Presentation.DTO
+{
+   public class OriginDataItemDTO
+   {
+      private readonly OriginDataItem _originDataItem;
+
+      public OriginDataItemDTO(OriginDataItem originDataItem)
```
Comment (msevestre) [inline src/MoBi.Presentation/DTO/OriginDataItemDTO.cs:9]: Why do you need a DTO for this?

File: src/MoBi.Presentation/DTO/OriginDataDTO.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      public IEnumerable<OriginDataItemDTO> AllDataItems => _originDataItems;
+      public ValueOriginDTO ValueOrigin { get; }
+
+      public OriginDataDTO(OriginDataItems originData)
+      {
+         _originData = originData;
+         _originDataItems = new List<OriginDataItemDTO>();
+         _originData.AllDataItems.Each(x => _originDataItems.Add(new OriginDataItemDTO(x)));
+         ValueOrigin = new ValueOriginDTO(originData);
+
```
Comment (msevestre) [inline src/MoBi.Presentation/DTO/OriginDataDTO.cs:22]: run resharper in this file

File: src/MoBi.Presentation/DTO/OriginDataDTO.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      private readonly OriginDataItems _originData;
+      private readonly List<OriginDataItemDTO> _originDataItems;
+
+      public IEnumerable<OriginDataItemDTO> AllDataItems => _originDataItems;
+      public ValueOriginDTO ValueOrigin { get; }
+
+      public OriginDataDTO(OriginDataItems originData)
+      {
+         _originData = originData;
+         _originDataItems = new List<OriginDataItemDTO>();
```
Comment (msevestre) [inline src/MoBi.Presentation/DTO/OriginDataDTO.cs:19]: you could inline cosntruction with line 11

File: src/MoBi.Presentation/DTO/OriginDataDTO.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Presentation.DTO
+{
+   public class OriginDataDTO
+   {
+      private readonly OriginDataItems _originData;
+      private readonly List<OriginDataItemDTO> _originDataItems;
+
+      public IEnumerable<OriginDataItemDTO> AllDataItems => _originDataItems;
```
Comment (msevestre) [inline src/MoBi.Presentation/DTO/OriginDataDTO.cs:13]: return IReadOnly

File: src/MoBi.Presentation/Mappers/IndividualBuildingBlockToIndividualBuildingBlockDTOMapper.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      {
+         _individualParameterToDTOMapper = individualParameterToDTOMapper;
+         _originDataToOriginDataDTOMapper = originDataToOriginDataDTOMapper;
+      }
+
+      public IndividualBuildingBlockDTO MapFrom(IndividualBuildingBlock individualBuildingBlock)
+      {
+         return new IndividualBuildingBlockDTO(individualBuildingBlock)
+         {
+            OriginDataDTO = _originDataToOriginDataDTOMapper.MapFrom(individualBuildingBlock.OriginData),
```
Comment (msevestre) [inline src/MoBi.Presentation/Mappers/IndividualBuildingBlockToIndividualBuildingBlockDTOMapper.cs:27]: I am not sure having the properties called DTO is necessary. This is somewhat redundant. You are in a dto already

File: src/MoBi.Presentation/DTO/OriginDataDTO.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      public ValueOriginDTO ValueOrigin { get; }
+
+      public OriginDataDTO(OriginDataItems originData)
```
Comment (msevestre) [inline src/MoBi.Presentation/DTO/OriginDataDTO.cs:16]: I am not sure you need this DTO either. isn't ORiginData Reaadonly already?

**Review by msevestre — COMMENTED** [review]
looks good. 
Not sure about the DTO that are the same as the original object. Maybe we'll need them later but it looks more like a YAGNI to me

**Pull request merged**